.github/workflows/build_and_test.yml

```diff
@@ -12971,4 +12971,4 @@ jobs:
       shell: cmd
       run: |
         cd OpenDDS\build
-        ctest -C Debug
+        ctest --output-on-failure -C Debug
```

Comment:
This should be handled by `CTEST_OUTPUT_ON_FAILURE` [here](https://github.com/OpenDDS/OpenDDS/blob/6d807de50b1e7c9614b9c691375d0446b2657335/.github/workflows/build_and_test.yml#L66). Is that not working?

Reply:
It is, the first few CI tests didn't seem to be outputting but was failing in a different way or something.

Reply:
This is a guess, but you might have been looking at MPC-built OpenDDS builds. In those CMake tests are still run using ctest, but using different logging and reporting mechanism that gets put into artifacts. The CMake-built OpenDDS builds just use ctest directly. It's on a TODO list somewhere to decide what to do about this difference.